File: services/ordering/src/main/java/io/wkrzywiec/fooddelivery/ordering/domain/outgoing/TipAddedToOrder.java
package io.wkrzywiec.fooddelivery.ordering.domain.outgoing;

import io.wkrzywiec.fooddelivery.commons.event.IntegrationMessageBody;

import java.math.BigDecimal;

public record TipAddedToOrder(String orderId, int version, BigDecimal tip, BigDecimal total) implements IntegrationMessageBody {
}

File: services/ordering/src/main/kotlin/io/wkrzywiec/fooddelivery/ordering/domain/outgoing/TipAddedToOrder.kt
package io.wkrzywiec.fooddelivery.ordering.domain.outgoing

import io.wkrzywiec.fooddelivery.commons.event.IntegrationMessageBody
import java.math.BigDecimal
import java.util.*

data class TipAddedToOrder(val id: UUID, val aggregateVersion: Int, val tip: BigDecimal, val total: BigDecimal) : IntegrationMessageBody {
    override fun orderId(): UUID = id
    override fun version(): Int = aggregateVersion
}
